### PR TITLE
Backports 0.15 pr11

### PR DESCRIPTION
**Pull request by PastaPastaPasta**
~~based on #2918~~
Also, btw, this doesn't build atm

### CODE REVIEW

**Comment by UdjinM6** [comment]
I propose to split this PR in 3 parts:
1. 10090-10186 (i.e. up to validation interface changes)
2. 9480-9942 (i.e. up to fee estimation changes)
3. 10181-9951 (i.e. the rest of it including wallet db changes)

File: contrib/debian/examples/dash.conf [hunk 1]
```diff
@@ -131,6 +131,13 @@
 # be validated sooner.
 #paytxfee=0.00
 
+# Enable pruning to reduce storage requirements by deleting old blocks. 
+# This mode is incompatible with -txindex and -rescan.
+# 0 = default (no pruning).
+# 1 = allows manual pruning via RPC.
+# >=550 = target to stay under in MiB.
```
Comment (UdjinM6) [inline contrib/debian/examples/dash.conf]: s/550/945/ see `MIN_DISK_SPACE_FOR_BLOCK_FILES` in `validation.h`

File: contrib/debian/examples/dash.conf [hunk 2]
```diff
@@ -131,6 +131,13 @@
 # be validated sooner.
 #paytxfee=0.00
 
+# Enable pruning to reduce storage requirements by deleting old blocks. 
+# This mode is incompatible with -txindex and -rescan.
+# 0 = default (no pruning).
+# 1 = allows manual pruning via RPC.
+# >=550 = target to stay under in MiB. 
+#prune=550
```
Comment (UdjinM6) [inline contrib/debian/examples/dash.conf]: same

File: test/util/data/bitcoin-util-test.json [hunk 3]
```diff
@@ -109,6 +109,38 @@
     "description": "Malformed outpubkey argument (too many separators). Expected to fail."
   },
   { "exec": "./dash-tx",
+    "args":
```
Comment (UdjinM6) [inline test/util/data/bitcoin-util-test.json]: 10130 should be dropped, was backported earlier

File: configure.ac [hunk 4]
```diff
@@ -626,6 +626,14 @@ AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[#include <malloc.h>]],
  [ AC_MSG_RESULT(no)]
 )
 
+dnl Check for mallopt(M_ARENA_MAX) (to set glibc arenas)
```
Comment (UdjinM6) [inline configure.ac]: 10120 should be dropped, was backported earlier

**Comment by PastaPastaPasta** [comment]
dropped 10020 and 10030 and changed dash.conf

**Comment by UdjinM6** [comment]
Pls also see https://github.com/UdjinM6/dash/commit/b382e671f696717fe2c8297e2aace8fef1beec23. I'm not yet 100% sure if it works correctly, still running tests https://travis-ci.org/UdjinM6/dash/builds/536320920. But at least it compiles locally.

File: src/instantsend.h [hunk 5]
```diff
@@ -148,7 +148,7 @@ class CInstantSend
     void Relay(const uint256& txHash, CConnman& connman);
 
     void UpdatedBlockTip(const CBlockIndex *pindex);
-    void SyncTransaction(const CTransaction& tx, const CBlockIndex *pindex, int posInBlock);
+    void SyncTransaction(const CTransactionRef& tx, const CBlockIndex *pindex = NULL, int posInBlock = 0);
```
Comment (PastaPastaPasta) [inline src/instantsend.h]: `NULL` vs `nullptr`?

Comment (UdjinM6) [reply src/instantsend.h]: https://github.com/UdjinM6/dash/commit/f3c800eb1501515709b575d7e5301605670e6b98

**Comment by UdjinM6** [comment]
ehh... `llmq-is-cl-conflicts.py` fails locally :/

**Comment by UdjinM6** [comment]
ok, let's try new version :) dfeffa09d12e797a325ad4881b2f72aeda62366b + d96c87859b390c9bc44d081193a37ffb538d730b, Travis https://travis-ci.org/UdjinM6/dash/builds/536378877 (make sure to drop a5aafd6 and 594e8a1 first).

**Comment by UdjinM6** [comment]
Pls see https://github.com/UdjinM6/dash/commit/25189b0e7e2c9fa20030a604b6b7e0c72d56102c, Travis https://travis-ci.org/UdjinM6/dash/builds/536765016

**Comment by codablock** [comment]
The whole backport of https://github.com/bitcoin/bitcoin/pull/10133 should be removed from this PR, as it was already backported as part of #1697. Later backported PRs found in #1701 then changed some of the values/calculations which are now basically reverted by this re-backporting of bitcoin#10133

**Comment by PastaPastaPasta** [comment]
removed 10133

**Pull request merged**